### PR TITLE
Fix some NumPy 2.x CI failures (cont.) 

### PR DESCRIPTION
**Pull request by EarlMilktea**
Related to #8565 .

This PR partly fixes CI failures marked with `XXX`.

### CODE REVIEW

File: cupy/fft/_fft.py [hunk 1]
```diff
@@ -253,6 +255,11 @@ def _fft(a, s, axes, norm, direction, value_type='C2C', overwrite_x=False,
         out_size = _get_fftn_out_size(a.shape, s, axes[-1], value_type)
         a = _exec_fft(a, direction, value_type, norm, axes[-1], overwrite_x,
                       out_size)
+        # Numpy compatibility
+        # 1D FFT: Regain the original type (before _convert_dtype)
+        # Others: Not necessary as numpy applies C2C iFFTs before R2C iRFFT
+        if len(axes) == 1:
+            a = a.astype(real_dtype)
```
Comment (asi1024) [inline cupy/fft/_fft.py:262]: For performance.
```suggestion
            a = a.astype(real_dtype, copy=None)
```

Comment (EarlMilktea) [reply cupy/fft/_fft.py]: Fixed in 0afe0cf7d266559e47260afb94c51a4279acbbfe .

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**